### DISCUSSION TITLE
Why suspend only pools whose ZIO_CHILD_LOGICAL type is incorrect?

### DISCUSSION THREAD

**Discussion by yuanfangcc** [openzfs/zfs#16235] [General]
In function ` zio_done` , when `if (zio->io_error && zio == zio->io_logical)` 
The pool will be suspended.
Why can't other types of zio be suspended?

**Comment by amotin** [comment]
Pool suspension is the last resort when some error can not be reasonably handled, or the handling is not implemented.  I'd say it is great that we can handle errors in other types of errors.  It is actually easier to do that for other ZIOs, since they usually don't have a problem to just report the error, while for some logical ZIOs the error is not a answer.  It would be good to improve error handling for logical ZIOs to less rely on suspension also.
